File: src/iBME/superceded_GS/cpp/stractNet/ShortFrag.cpp
///////////////////////////////////////////////////////////
//  ShortFrag.cpp
//  Implementation of the Class ShortFrag
//  Created on:      06-Aug-2008 7:51:59 AM
//  Original author: david.h.loman
///////////////////////////////////////////////////////////
#include "ShortFrag.h"
///////////////////////////////////////////////////////////////////////
/*
 *
 * xSTRUCTORS
 *
 */
///////////////////////////////////////////////////////////////////////
ShortFrag::~ShortFrag(){}
ShortFrag::ShortFrag(Short sh){
	MsgFrag("ShortFrag", sh, MsgFrag::mftShort);
}
ShortFrag::ShortFrag(DataInputStream& in){
	MsgFrag("ShortFrag", in, MsgFrag::mftShort);
}
///////////////////////////////////////////////////////////////////////
/*
 *
 * METHODS
 *
 */
///////////////////////////////////////////////////////////////////////
void ShortFrag::Deserialize(DataInputStream& in){
	try {
		// Read Short
		this->field = in.readShort();
	} catch (IOException ioe) {
		SNRoot::err("IOException in Deserialize()", 0);
	}
	return;
}
void ShortFrag::Serialize_Specific(DataOutputStream& out){
	try {
		// write in new data
		out.writeShort(this->field);
	} catch (IOException ioe) {
		SNRoot::err("IOException in Serialize()", 0);
	}
	return;
}

File: src/superceded_GS/java/stractNet/src/stractNet/messaging/msgs/frags/ShortFrag.java
package stractNet.messaging.msgs.frags;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;

public class ShortFrag extends MsgFrag<Short> {
	public ShortFrag(Short sh) {
		super("ShortFrag", sh, MsgFrag.mftShort);
	}

	public ShortFrag(DataInputStream in) {
		super("ShortFrag", in, MsgFrag.mftShort);
	}

	@Override
	public void Deserialize(DataInputStream in) {
		try {
			// Read Short
			this.field = in.readShort();
		} catch (IOException ioe) {
			this.err("IOException in Deserialize()", 0);
		}
		return;
	}

	@Override
	protected void Serialize_Specific(DataOutputStream out) {
		try {
			// write in new data
			out.writeShort(this.field);
		} catch (IOException ioe) {
			this.err("IOException in Serialize()", 0);
		}

		return;
	}

}
